obspy/core/tests/test_util_base.py

```diff
@@ -90,6 +93,24 @@ def test_image_comparison(self):
         # check that temp file is deleted
         self.assertFalse(os.path.exists(ic.name))
 
+    def test_mock_read_inventory_http_errors(self):
+        """
+        Tests HTTP Error on 204, 400, and 500
+        """
+        url = "http://obspy.org"
+        for response_tuple in [("204", "No Content"), ("400", "Bad Request"),
+                               ("500", "Internal Server Error")]:
+            code = response_tuple[0]
+            reason = response_tuple[1]
+            with mock.patch("requests.get") as mocked_get:
+                mocked_get.return_value.status_code = code
+                mocked_get.return_value.reason = reason
+                with self.assertRaises(HTTPError) as e:
+                    download_to_file(url, None)
+                    self.assertEqual(e.exception.args[0],
+                                     "%s HTTP Error: %s for url: %s" %
+                                     (code, reason, url))
```

Comment:
Can you move the asserting statement out of the context manager? In this case it probably does not matter but the `assertRaises()` would also be satisfied if the exception is raised in the `assertEqual()` call if it is part of the context manager.